modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -193,65 +195,108 @@ public CompletableFuture<Void> resetPartitions(int zoneId, int tableId) {
+    public CompletableFuture<Map<TablePartitionId, GlobalPartitionState>> globalPartitionStates(
+            Set<String> zoneNames,
+            Set<Integer> partitionIds
+    ) {
         Catalog catalog = catalogManager.catalog(catalogManager.latestCatalogVersion());
 
-        return localPartitionStatesInternal(zoneName, catalog)
+        return localPartitionStatesInternal(zoneNames, Set.of(), partitionIds, catalog)
+                .thenApply(res -> checkPartitions(res, partitionIds))
                 .thenApply(res -> normalizeLocal(res, catalog))
                 .thenApply(res -> assembleGlobal(res, catalog));
     }
 
     private CompletableFuture<Map<TablePartitionId, Map<String, LocalPartitionStateMessage>>> localPartitionStatesInternal(
-            @Nullable String zoneName, Catalog catalog
+            Set<String> zoneNames,
+            Set<String> nodeNames,
+            Set<Integer> partitionIds,
+            Catalog catalog
     ) {
-        int zoneId;
-        if (zoneName == null) {
-            zoneId = NO_ZONE_ID;
+        Set<Integer> zoneIds;
+        if (zoneNames.isEmpty()) {
+            zoneIds = Set.of();
         } else {
-            Optional<CatalogZoneDescriptor> zoneDesciptorOptional = catalog.zones().stream()
-                    .filter(catalogZoneDescriptor -> catalogZoneDescriptor.name().equals(zoneName))
-                    .findAny();
+            Set<String> uppercaseZoneNames = zoneNames.stream()
+                    .map(String::toUpperCase)
+                    .collect(toSet());
+
+            List<CatalogZoneDescriptor> zoneDescriptors = catalog.zones().stream()
+                    .filter(catalogZoneDescriptor -> uppercaseZoneNames.contains(catalogZoneDescriptor.name().toUpperCase()))
+                    .collect(toList());
+
+            if (zoneDescriptors.size() != zoneNames.size()) {
+                Set<String> foundZoneNames = zoneDescriptors.stream()
+                        .map(descriptor -> descriptor.name().toUpperCase())
+                        .collect(toSet());
 
-            if (zoneDesciptorOptional.isEmpty()) {
-                return CompletableFuture.failedFuture(new DistributionZoneNotFoundException(zoneName, null));
+                Set<String> missingZones = CollectionUtils.difference(uppercaseZoneNames, foundZoneNames);
+
+                return CompletableFuture.failedFuture(new ZonesNotFoundException(missingZones));
             }
 
-            CatalogZoneDescriptor zoneDescriptor = zoneDesciptorOptional.get();
-            zoneId = zoneDescriptor.id();
+            zoneIds = zoneDescriptors.stream()
+                    .map(CatalogObjectDescriptor::id)
+                    .collect(toSet());
         }
 
-        Set<NodeWithAttributes> logicalTopology = dzManager.logicalTopology();
+        Set<String> uppercaseNodeNames = nodeNames.stream()
```

Comment:
Again, is it a rule that you capitalize node names?
Let's make the code related to searching for nodes a separate method.